addons/web/controllers/main.py

```diff
@@ -451,6 +423,178 @@ def _admin_password_warn(uid):
             subtype='mail.mt_comment'
         )
 
+
+class HomeStaticTemplateHelpers(object):
+    """
+    Helper Class that wraps the reading of static qweb templates files
+    and xpath inheritance applied to those templates
+    /!\ Template inheritance order is defined by ir.module.module natural order
+        which is "sequence, name"
+        Then a topological sort is applied, which just puts dependencies
+        of a module before that module
+    """
+    NAME_TEMPLATE_DIRECTIVE = 't-name'
+    STATIC_INHERIT_DIRECTIVE = 't-inherit'
+    STATIC_INHERIT_MODE_DIRECTIVE = 't-inherit-mode'
+    STATIC_INHERIT_MODES = ['primary', 'extension']
+    DEFAULT_MODE = 'primary'
+
+    def __init__(self, addons, db, checksum_only=False, debug=False):
+        super(HomeStaticTemplateHelpers, self).__init__()
+        self.addons = addons.split(',') if isinstance(addons, str) else addons
+        self.db = db
+        self.debug = debug
+        self.checksum_only = checksum_only
+        self.template_dict = OrderedDict()
+
+    def _get_parent_coordinates(self, addon, template):
+        """Computes the real addon name and the template name
+        of the parent template (the one that is inherited from)
+
+        :param: string; addon, the addon the template is declared in
+        :param: etree node: the current template we are are handling
+        :returns (str, str, str)
```

Comment:
The name of the method "coordinates" doesn't match with the comment.
Also, I don't understand what the "real addon name" is